backend/inspirehep/records/api/mixins.py

```diff
@@ -114,242 +112,119 @@ def update_refs_in_citation_table(self, save_every=100):
 
 
 class FilesMixin:
-    def add_file(self, url, original_url=None, filename=None, **kwargs):
-        """Downloads file from url and saves it with `filename` as a proper name.
-
-        If filename is not provided ti will be resolved in the following way:
-            - Use filename if it's provided
-            - Check if key looks as proper filename, if yes then use it.
-            - if original_url is provided and last part of it (after last `/`)
-              looks like proper filename then use it
-            - as last resort use last part of `url` (after last `/`)
-
-        Args:
-            url (string): Url to a file, or to local api
-            filename (string): Proper name of the file.
-            original_url (string): URL from which this document was downloaded
-
-        Keyword Args:
-            description (string): works for documents and figures
-            fulltext (bool): works for documents only
-            hidden (bool): works for documents only
-            material (string): works for documents and figures
-            caption (string): works for facets only
-            key (string): Can contain name of the file (compatibility with inspire-next)
+    @classmethod
+    def create_bucket(cls, data):
+        location = current_app.config["RECORDS_DEFAULT_FILE_LOCATION_NAME"]
+        storage_class = current_app.config["RECORDS_DEFAULT_STORAGE_CLASS"]
+        return Bucket.create(location=location, storage_class=storage_class)
 
-        Returns:
-            dict: Metadata for file
-        """
-        key = self.find_and_add_file(url, original_url)
-        if not key:
-            raise FileNotFoundError(f"File `{url}|{original_url}` not found")
+    @property
+    def files(self):
+        if self.model is None:
+            raise MissingModelError()
+        bucket = self.get_bucket()
+        return self.files_iter_cls(self, bucket=bucket, file_cls=self.file_cls)
+
+    def add_file(self, url, original_url=None, key=None, filename=None, **kwargs):
+        if self.local_url(url):
+            bucket, key = self.find_bucket_and_key_from_local_url(url)
+            key = self.add_local_file(key, bucket)
+        else:
+            key = self.add_file_from_url(url)
```

Comment:
You are not handling situation when local url will be broken and you have to re-download file form original_url (this is why we decided to store original url) - requirement from files refactoring

Reply:
if there is local url then we copy from the storage, what's the point to download it from the original url ? 😌

Reply:
I do not remember now but It was one of the requirements on the original files refactor.

Reply:
I see 2 use-caces here. 1. File updated on original url, 2. file is broken on local storage.